ibc-core/ics02-client/types/src/status.rs

```diff
@@ -53,6 +53,14 @@ impl Status {
             status => Err(ClientError::ClientNotActive { status }),
         }
     }
+
+    /// Checks whether the client is either frozen or expired; returns `Err` if not.
+    pub fn verify_is_inactive(self) -> Result<(), ClientError> {
+        match self {
+            Self::Frozen | Self::Expired => Ok(()),
+            status => Err(ClientError::ClientNotInactive { status }),
```

Comment:
Can we be exhaustive here? If we add more enum variants, they will be matched to the second match. Note `Status::Unauthorized` gets matched for `NotInactive` case.

Or should be make it

```
            Self::Active => Err(ClientError::ClientNotInactive { status }),
            _ => Ok(()),
```

Reply:
I'm not sure there's a benefit to being exhaustive here. According to the client recovery spec:
> The first client identifier is the proposed client to be updated. This client must be either frozen or expired.

So it sounds like we only need to explicitly check for a frozen or expired status. To be honest, I'm not sure how an unauthorized client should be treated as far as client recovery is concerned, so having it return an error seemed prudent.

Reply:
Looks good in that case :+1: